### PR TITLE
Crash on Telegram bot errors

### LINKED ISSUE #15: teleirc does not "crash" if it encounters a fatal error
teleirc does not crash and return an exit code if it encounters an unrecoverable error. Rather, it fails and hangs without exiting.

### PR DESCRIPTION
**Pull request by thenaterhood**
This may resolve #15 and #67 but since I can't replicate the exact scenario for those, I can't make guarantees.

**WARNING** this updates the version of the telegram bot library in use, which is a requirement for these changes to work.

### CODE REVIEW

**Comment by Tjzabel** [comment]
Hm, CodeClimate has been super pedantic lately. @thenaterhood thanks for the PR! @jwflory what do you think?

**Comment by Tjzabel** [comment]
I think we should keep this PR open until [PR #72](https://github.com/RITlug/teleirc/pull/72) goes through.
The location of the error catching may need to be changed depending on this PR going through.

**Comment by thenaterhood** [comment]
AFAIK #72 only touches the IRC side not the Telegram side. Once the Telegram side gets a similar refactor then the error handling code could move.

Code climate has configurable thresholds. We could decide on what thresholds are reasonable for teleirc and add a config file to make certain checks less pedantic. The defaults are here: https://docs.codeclimate.com/docs/advanced-configuration#section-default-checks

**Comment by Tjzabel** [comment]
@thenaterhood it seems #72 does only touch the IRC side. There are still many items taken out of the TeleIrc.js file, but I suppose this won't affect anything with this PR.

And yes, I do think after seeing the past few PRs, that it may be a good idea to dial down CodeClimate's pedantic level.

**Comment by xforever1313** [comment]
Changes look fine to me.  They shouldn't interfere with #72, as this only changes teleirc.js, while #72 doesn't even touch that file.

**Comment by jwflory** [comment]
@thenaterhood #65 is now merged. Could you please rebase your changes on top of the `HEAD` of master branch? I'm not sure how `git rebase` savvy you are, but there is an excellent [blog post guide](https://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/) and a [edX tutorial](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request) on rebasing that might help.

Alternatively, we could wait for #72 to get merged before rebasing, but I see @xforever1313 noting that the changes are independent of each other.

**Comment by thenaterhood** [comment]
Rebased :fireworks:

File: teleirc.js [hunk 1]
```diff
@@ -28,6 +28,25 @@ let ircbot = new irc.Client(config.irc.server, config.irc.botName, {
...
 let tgbot = new tg(config.token, { polling: true });
+// This is super rudimentary error handling. If the Telegram bot
+// throws an error, the library we use simply logs it and appears
+// to hang. This may be a bug in that library (see
+// https://github.com/yagop/node-telegram-bot-api/issues/657).
+// This code forces teleirc to fully crash when it encounters a polling
+// error so that its parent process/container can restart it.
+// It may be possible to handle this in a nicer way, but issues
+// open on the bot library indicate that things may not be working right
+// as of October 2018.
```
Comment (jwflory) [inline teleirc.js]: Thanks for the context explanation here. :+1:

**Pull request merged**